kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs

```diff
@@ -690,12 +697,15 @@ fn new_machine_model(sess: &Session) -> MachineModel {
             let double_width = 64;
             let float_width = 32;
             let int_width = 32;
-            let long_double_width = 64;
+            let long_double_width = match os.as_ref() {
+                "linux" => 128,
+                _ => 64,
+            };
             let long_int_width = 64;
             let long_long_int_width = 64;
             let short_int_width = 16;
             let single_width = 32;
-            let wchar_t_is_unsigned = false;
+            let wchar_t_is_unsigned = matches!(os.as_ref(), "linux");
```

Comment:
Any reference we can cite?

Reply:
It's quite hard find (and link) this information in ARM manuals, which is why I'd suggest the first two ([here](https://developer.arm.com/documentation/dui0491/i/Compiler-Command-line-Options/--signed-chars----unsigned-chars) and [here](https://www.arm.linux.org.uk/docs/faqs/signedchar.php)) for Linux and the last one ([here](https://developer.apple.com/documentation/xcode/writing-arm64-code-for-apple-platforms)) for Apple.
```suggestion
            // https://developer.arm.com/documentation/dui0491/i/Compiler-Command-line-Options/--signed-chars----unsigned-chars
            // https://www.arm.linux.org.uk/docs/faqs/signedchar.php
            // https://developer.apple.com/documentation/xcode/writing-arm64-code-for-apple-platforms
            let wchar_t_is_unsigned = matches!(os.as_ref(), "linux");
```
What do you think?

Reply:
Looks good. Thanks for digging up the references!